harness/determined/keras/_tf_keras_trial.py

```diff
@@ -746,50 +759,71 @@ def _launch_evaluate(self) -> Any:
         return metrics
 
     def _control_loop(self) -> None:
-        for wkld, args, response_func in self.workloads:
-            logging.debug(f"Received wkld {wkld.kind} with args {args}.")
+        assert self.workloads is not None
+        for wkld, response_func in self.workloads:
+            start_time = self._generic._current_timestamp()
+            logging.debug(f"Received wkld {wkld.kind}.")
+
             if wkld.kind == workload.Workload.Kind.RUN_STEP:
                 # Configure the state for a training step.
+                self.train_workload = wkld
+                self.train_start_time = start_time
                 self.train_response_func = response_func
                 self.train_workload_batches = 0
                 self.train_workload_metrics = []
                 self.train_workload_len = wkld.num_batches
                 self.multiplexer.set_batches_requested(wkld.num_batches)
-                break
+                return
+
             elif wkld.kind == workload.Workload.Kind.COMPUTE_VALIDATION_METRICS:
                 try:
-                    response_func(
-                        det.util.wrap_metrics(
-                            self._compute_validation_metrics(),
-                            self.context.get_stop_requested(),
-                            invalid_hp=False,
-                            init_invalid_hp=False,
-                        )
+                    response = det.util.wrap_metrics(
+                        self._compute_validation_metrics(),
+                        self.context.get_stop_requested(),
+                        invalid_hp=False,
+                        init_invalid_hp=False,
                     )
                 except det.InvalidHP as e:
                     logging.info(
                         "Invalid hyperparameter exception in trial validation step: {}".format(e)
                     )
-                    response_func(
-                        util.wrap_metrics(
-                            {},
-                            self.context.get_stop_requested(),
-                            invalid_hp=True,
-                            init_invalid_hp=False,
-                        )
+                    response = util.wrap_metrics(
```

Comment:
there's `util.wrap_metrics` here but `det.util.wrap_metrics` in line 780

Reply:
I'd actually like to get rid of `workload.Skipped` entirely, since there's no longer the same SocketManager layer that used to consume that value.  Getting rid of `workload.Skipped` means I'd also like to remove `util.wrap_metrics()`, so this is no longer relevant.